plugins/nv-filters/nvidia-videofx-filter.c

```diff
@@ -359,229 +707,138 @@ static bool process_texture_greenscreen(struct nv_greenscreen_data *filter)
-					  obs_source_t *context)
-{
-	struct nv_greenscreen_data *filter =
-		(struct nv_greenscreen_data *)bzalloc(sizeof(*filter));
-	if (!nvvfx_loaded) {
-		nv_greenscreen_filter_destroy(filter);
-		return NULL;
-	}
-
-	NvCV_Status vfxErr;
-	filter->context = context;
-	filter->mode = -1; // should be 0 or 1; -1 triggers an update
-	filter->images_allocated = false;
-	filter->processed_frame = true; // start processing when false
-	filter->width = 0;
-	filter->height = 0;
-	filter->initial_render = false;
-	os_atomic_set_bool(&filter->processing_stop, false);
-	filter->handler = NULL;
-	filter->processing_interval = 1;
-	filter->processing_counter = 0;
-
-	/* 1. Create FX */
-	vfxErr = NvVFX_CreateEffect(NVVFX_FX_GREEN_SCREEN, &filter->handle);
-	if (NVCV_SUCCESS != vfxErr) {
-		const char *errString = NvCV_GetErrorStringFromCode(vfxErr);
-		error("Error creating AI Greenscreen FX; error %i: %s", vfxErr,
-		      errString);
-		nv_greenscreen_filter_destroy(filter);
-		return NULL;
-	}
-
-	/* 2. Set models path & initialize CudaStream */
-	char buffer[MAX_PATH];
-	char modelDir[MAX_PATH];
-	nvvfx_get_sdk_path(buffer, MAX_PATH);
-	size_t max_len = sizeof(buffer) / sizeof(char);
-	snprintf(modelDir, max_len, "%s\\models", buffer);
-	vfxErr = NvVFX_SetString(filter->handle, NVVFX_MODEL_DIRECTORY,
-				 modelDir);
-	vfxErr = NvVFX_CudaStreamCreate(&filter->stream);
-	if (NVCV_SUCCESS != vfxErr) {
-		const char *errString = NvCV_GetErrorStringFromCode(vfxErr);
-		error("Error creating CUDA Stream; error %i: %s", vfxErr,
-		      errString);
-		nv_greenscreen_filter_destroy(filter);
-		return NULL;
-	}
-	vfxErr = NvVFX_SetCudaStream(filter->handle, NVVFX_CUDA_STREAM,
-				     filter->stream);
-	if (NVCV_SUCCESS != vfxErr) {
-		const char *errString = NvCV_GetErrorStringFromCode(vfxErr);
-		error("Error setting CUDA Stream %i", vfxErr);
-		nv_greenscreen_filter_destroy(filter);
-		return NULL;
-	}
-	/* check sdk version */
-	if (NvVFX_GetVersion(&filter->version) == NVCV_SUCCESS) {
-		uint8_t major = (filter->version >> 24) & 0xff;
-		uint8_t minor = (filter->version >> 16) & 0x00ff;
-		uint8_t build = (filter->version >> 8) & 0x0000ff;
-		uint8_t revision = (filter->version >> 0) & 0x000000ff;
-		// sanity check
-		nvvfx_new_sdk = filter->version >= MIN_VFX_SDK_VERSION &&
-				nvvfx_new_sdk;
-	}
-
-	/* 3. Load alpha mask effect. */
-	char *effect_path = obs_module_file("rtx_greenscreen.effect");
-
-	obs_enter_graphics();
-	filter->effect = gs_effect_create_from_file(effect_path, NULL);
-	bfree(effect_path);
-	if (filter->effect) {
-		filter->mask_param =
-			gs_effect_get_param_by_name(filter->effect, "mask");
-		filter->image_param =
-			gs_effect_get_param_by_name(filter->effect, "image");
-		filter->threshold_param = gs_effect_get_param_by_name(
-			filter->effect, "threshold");
-		filter->multiplier_param = gs_effect_get_param_by_name(
-			filter->effect, "multiplier");
+	/* 3. run AIGS fx */
```

Comment:
```suggestion
	/* 3. Run AIGS FX */
```

Reply:
fixed and I added the explanation for AIGS = AI Greenscreen